Update French translation

rbazaud opened the pull request:
File config/translations/lxqt-config-globalkeyshortcuts_fr_FR.ts

palinek commented on config/translations/lxqt-config-globalkeyshortcuts_fr_FR.ts:
```diff
@@ -49,47 +49,47 @@
         <location filename="../edit_action_dialog.ui" line="118"/>
         <source>Type:</source>
-        <translation type="unfinished"></translation>
+        <translation>Type :</translation>
```
Is the spaces before double colon intentional? (also in all other places)

rbazaud replied:
Yes, it is, in French typography it is mandatory to put a white space in front of colons, exclamation and interrogation points.

pmattern commented:
If I'm not mistaken `Enabled` in line 66 is referring to shortcut / raccourci only so `Activé` will do, IMO.

While at it I saw that quite a lot of preexisting stuff is faulty or missing.
Many strings aren't translated at all. Lines 125, 155 and 165 were translated without removing tag `type="unfinished"`. In several strings the ampersand defining a mnemonic is missing.
You may want to consider fixing this in addition. Can be done rather easily with QtLinguist, FWIW. Regarding the Git stuff you may ask us if needed.

rbazaud commented:
Ok, I thought it referred to the «action» but I stand corrected. Thanks for the advice, I will give it a try (Qt Linguist) though I'm not sure I can run it right now (memory troubles). Is it ok to do changes and cleanup in the same PR ? I mean in different commits, obviously. And about the mnemonics, do they rather have to be the same for both languages or doesn’t it matter ?

pmattern commented:
I'd really recommend you to try QtLinguist as it makes things a lot easier and helps avoiding errors at the same time.
If you've got hardware trouble right now just take your time. We're not in a hurry.

As for translations it's absolutely okay to mix different kind of changes like first-time translations and error corrections in one PR. It's even better to put all in a single commit as several commits covering different aspects of translation of a single file just mean unnecessary clutter in Git history which is something we want to avoid. It's just that the commit message should be appropriate - which definitely applies to the one you've chosen already.

There's no need to keep the mnemonics as in English. In fact frequently this isn't possible at all and chosing different ones is way more intuitive in the respective language. The only thing that matters is avoiding duplicates. Again QtLinguist is helpful as it's providing screenshots of the GUI you're dealing with.

rbazaud commented:
Thank you very much for this detailed answer. The file being not that large, I could double-check every mnemonic one-by-one for collisions, but I definitely see your point about the interest of Qt Linguist with regard to duplicates and reliability.

All right, I will squash the commits then. I have learned it to be better not to mix changes and cleanups for the code, then I preferred to ask before doing so with translations.

rbazaud commented:
@pmattern : could you make a quick review of the last version please ?

pmattern commented:
In context "EditActionDialog" mnemonics D, E and M are is assigned twice (to my knowledge there's no difference between upper and lower case if it comes to mnemonics). Don't know where the last four strings Service, Path, ... are meant to appear in the first place, though.
In context "MainWindow" two strings aren't translated.
Aside from this the actual translation is alright.

But there's one other more general thing I missed so far - the question about need for a *country* specific translation. It's something that comes up again and again, maybe we should have a wiki page about it.
In some languages like Portuguese differences between countries exist that make country specific variants necessary. E. g. in German they aren't. The language gets spoken in several countries and the variants are quite different. But the technical terms which are what matters here there are no differences which urge country specific variants like `de_AT`, `de_CH` or `de_DE` and just `de` is sufficient.
Not sure about the situation in French. I'd basically assume it's pretty similar than in German.
Technically it's completely all right to have the language only. E. g. a translation `de` gets picked up on a system with locale `de_DE` as well.

rbazaud commented:
Thanks for the review. About the “_case-sensitiveness_” of mnemonics I currently don't know how `Qt` handles it. The previous translations weren’t from me, and about the English source I assumed that they were correct. Some technical terms e.g. DBUS aren’t to be translated, which allowed exceptionally to keep the same mnemonics.

About the missing translations, I honestly don't know how to translate them, I mean it's one thing to get the general idea or to have a few words and find an exact match which doesn't betray nor limit the meaning intended.

About regional settings for a common language, there are indeed several variations to French e.g. `fr_CA` for Canada (more precisely _Québec_, for instance they don’t have the same word for «browser» as ourself), `fr_CH` for Switzerland (or rather a part of it), … but I don't know all of them. As right now there is only one of these variations I prefer to keep it simple. If someone wants to contribute to `fr_CA` it will always be possible to split then.

rbazaud commented:
@pmattern mnemonics have been «fixed» so that there isn’t any collision. About the missing translations though, I still don't have a better idea.

pmattern commented:
It's completely fine to have a single translation only for the time being even if different country specific variants would make sense (which clearly applies to French given what you wrote about Québec). But it has to be named according to the language then, here `fr`, both the filename and that header in line 3. Btw. the latter gets set by QtLinguist by string "Any Country" from Edit - Translation File Settings.
A translation `fr` gets applied if any of `fr_{BE,CH,FR}` are set as system locale but lack a translation on their own, but a translation `fr_FR` does *not* get applied if the system locale is set to e. g. `fr_CH`.

Disagree about the double assignment of mnemonic 'D'. It should be possible to select all items covered by mnemonics by a character on their own.
Not a bummer, of course.

"Swap" is used in the meaning "exchange" here, donc peut-être échanger?
And "Global Actions Manager". Yeah. It's something that's always been bothering me, just didn't report yet. What the heck is this meant to be? I mean, if you have no clue and I have no clue, how should we expect users to have a clue? I think just translate it as "LXQt keyboard shortcuts" and blame it to me if someone complains... (Seriously, this is something that needs to get changed in English in the long run, if you ask me.)

rbazaud commented:
:open_mouth: right I didn’t see the reference to `fr_FR`, it is now fixed.

About the duplicated mnemonics `&amp;D`, it appears I made a mistake by changing the source string. it shouldn’t have been done, anyway not here, and not in a PR dedicated to French translation which is why I reverted back on it.

About «swap», I’ll go with «_échanger_» then, I didn't used it since I figured there might be some subtle difference with «exchange» but you’re right it shouldn’t confuse the users as it is, given the context.

For the «Global Actions Manager» I used the translation you suggested i.e. the French version of «LXQt keyboard shortcuts» : «_Raccourcis clavier LXQt_». Not only is it much more clearer but it is also closer to the name in the menu (defined by a `.desktop` file I think) which is currently «Shortcut Keys»

pmattern commented:
Translation itself is alright now, IMO.
You *could* consider adding some translator comment regarding "Global Actions Manager", something like "not literally at all but seems more reasonable like this" or so.

But your forgot to change the filename. Must be `_fr` only, too. `git mv` is your friend here.

rbazaud commented:
Ok, great. I did consider adding a translator comment, in fact I even did it :smile: (with Qt Linguist): «Not exactly the same wording but it seems more reasonable like this»

Thanks for the tip, I was going for `git add` / `git rm` (and `/bin/mv`) but `git mv` is more efficient.

pmattern commented:
`git mv` is just an automatic invocation of the very three commands you stated for the lazy ones.

That aside everything is fine now. We've been discussing so extensively that I think it's justified to merge without waiting for yet another review.
You may decide whether you want to do it yourself as you recently said you're looking for some Git practice. Should be rather easy as there haven't been any commits in `master` since you branched `update-fr-translation`. Just checkout into your local copy of the latter, merge upstream master and push.

rbazaud commented:
@pmattern :smile: lazy is ok for now.

PR merged and (automatically) closed. Once again, thank you for your help. I did push to origin (my repo) first then to upstream using a very convenient git alias (from @jleclanche ) for the merge part.

gh could provide the ability to disable the merge button (or make it hidden) for projects that don't use it, that would avoid a few mistakes…
